### PR TITLE
Feature/associations

### PR DESCRIPTION
**Pull request by assadounto**
### In this feature

- [x] I added the association for teacher and student in a classroom
- [x] I added association for books and rentals for students

### CODE REVIEW

File: objects_classes/book.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class Book
+  attr_accessor :title, :author, :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
```
Comment (cwaku) [inline objects_classes/book.rb]: - [x] There should be a `many-to-many` (also `has-many-through`) relationship between `Person` and `Book` using the intermediate class `Rental`. You did implement the association where a rental belongs to a book and also belongs to a person but not where a book and a person have many rentals.
  - [ ] Kindly add a method for adding rentals -the book itself should be sent as a parameter to create the new rental achieving the `has-many` association.

File: objects_classes/person.rb [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
+  attr_accessor :name, :age, :id, :parent_permission, :rentals
 
   def initialize(age, name = 'Unknown', parent_permission: true)
     @name = name
     @age = age
     @id = Random.rand(1..100)
     @parent_permission = parent_permission
+    @rentals = []
     super()
   end
```
Comment (cwaku) [inline objects_classes/person.rb:14]: - [ ] I suggest you add a method for adding rentals -the book itself should be sent as a parameter to create the new rental achieving the `has-many` association.

**Review by cwaku — CHANGES_REQUESTED** [review]
Hi @assadounto 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!
There are a few issues 🐛  that you still need to work 🛠️  on to go to the next project ⏭️, but you are almost there! 🏃🏾 

## Highlights
- [x] Clean/Organised code 👍🏾 
- [x] Linters passed 💯 
- [x] Good implementation 👍🏾 


## Required Changes ♻️
It's ok to have bugs in your code. Please, see this code review as a chance to learn practical coding skills and new tips 🌟  that can improve your code quality 💯 and help you get a better job 💼.

I will guide 📔 you through fixing 🛠️ these bugs, follow my comments down below and make sure you don't miss any of them to get your project approved ✅

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid, unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by roshan-bajgain — APPROVED** [review]
Hi @assadounto, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- No linter errors. ✔️
- GitHub flow is followed. ✔️
- PR has a good title and summary. ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**